app/scripts/metamask-controller.js

```diff
@@ -206,13 +209,15 @@ export default class MetamaskController extends EventEmitter {
       encryptor: opts.encryptor || undefined,
     })
     this.keyringController.memStore.subscribe((s) => this._onKeyringControllerUpdate(s))
+    this.keyringController.on('unlock', () => this.emit('unlock'))
```

Comment:
Could `this._onUnlock` be called directly here rather than re-emitting it on the controller itself? It can be a bit confusing tracking all of the events going around so if we can minimize those it would be nice.

Reply:
Oh I see - it was done here because `keyringController` is constructed after the app state controller. 🤔

Reply:
Yeah, I think at least one of the commits with failed tests is from me trying to do otherwise.